bin/MorphingMSSMvsSM.cpp

```diff
@@ -638,72 +733,81 @@ int main(int argc, char **argv) {
-        cb.cp().channel({chn}).process({"ggH1_b", "ggH2_b", "ggH3_b"}).ExtractShapes(
-          input_file_base, "$BIN/ggH_b_$MASS", "$BIN/ggH_b_$MASS_$SYSTEMATIC");
+      if(analysis == "bsm-model-dep-full"){
+        if(sm_like_hists == "bsm"){
+          if(sub_analysis == "sm-like-light" || sub_analysis == "sm-like-heavy"){
+            cb.cp().channel({chn}).process(mssm_ggH_signals_smlike).ExtractShapes(
+              input_file_base, "$BIN/$PROCESS_$MASS", "$BIN/$PROCESS_$MASS_$SYSTEMATIC");
+          }
+          else if(sub_analysis == "cpv"){
+            cb.cp().channel({chn}).process({"ggH1_t"}).ExtractShapes(
+              input_file_base, "$BIN/ggh_t_$MASS", "$BIN/ggh_t_$MASS_$SYSTEMATIC");
+            cb.cp().channel({chn}).process({"ggH1_b"}).ExtractShapes(
+              input_file_base, "$BIN/ggh_b_$MASS", "$BIN/ggh_b_$MASS_$SYSTEMATIC");
+            cb.cp().channel({chn}).process({"ggH1_i"}).ExtractShapes(
+              input_file_base, "$BIN/ggh_i_$MASS", "$BIN/ggh_i_$MASS_$SYSTEMATIC");
+          }
+          cb.cp().channel({chn}).process(mssm_bbH_signals_smlike).ExtractShapes(
+            input_file_base, "$BIN/bbH_$MASS", "$BIN/bbH_$MASS_$SYSTEMATIC");
+        }
+        else if(sm_like_hists == "sm125"){
+          cb.cp().channel({chn}).process(mssm_ggH_signals_smlike).ExtractShapes(
+            input_file_base, "$BIN/ggH125$MASS", "$BIN/ggH125$MASS_$SYSTEMATIC");
+          cb.cp().channel({chn}).process(mssm_bbH_signals_smlike).ExtractShapes(
+            input_file_base, "$BIN/bbH_125$MASS", "$BIN/bbH_125$MASS_$SYSTEMATIC");
+        }
+        cb.cp().channel({chn}).process(qqh_bsm_signals).ExtractShapes(
+          input_file_base, "$BIN/qqH125$MASS", "$BIN/qqH125$MASS_$SYSTEMATIC");
+        cb.cp().channel({chn}).process(wh_bsm_signals).ExtractShapes(
+          input_file_base, "$BIN/WH125$MASS", "$BIN/WH125$MASS_$SYSTEMATIC");
+        cb.cp().channel({chn}).process(zh_bsm_signals).ExtractShapes(
+          input_file_base, "$BIN/ZH125$MASS", "$BIN/ZH125$MASS_$SYSTEMATIC");
 
-        // Included in ggH1_x ?
-        // cb.cp().channel({chn}).process({"ggh"}).ExtractShapes(
-        //   input_file_base, "$BIN/ggH125$MASS", "$BIN/ggH125$MASS_$SYSTEMATIC");
-        cb.cp().channel({chn}).process(ch::JoinStr({sm_signals,main_sm_signals})).ExtractShapes(
+        // Adding SM125 signal templates for SM hypothesis
+        cb.cp().channel({chn}).process(ch::JoinStr({sm_signals, main_sm_signals})).process({"bbH125"}, false).ExtractShapes(
           input_file_base, "$BIN/$PROCESS$MASS", "$BIN/$PROCESS$MASS_$SYSTEMATIC");
-        cb.cp().channel({chn}).process({"qqH1"}).ExtractShapes(
-          input_file_base, "$BIN/qqH125$MASS", "$BIN/qqH125$MASS_$SYSTEMATIC");
+        cb.cp().channel({chn}).process({"bbH125"}).ExtractShapes(
+          input_file_base, "$BIN/bbH_125$MASS", "$BIN/bbH_125$MASS_$SYSTEMATIC"); // "bbH125" needs special treatment because of template name spelling
       }
+    }
   }
 
```

Comment:
TODO: Need to scale the bbH125 signal contribution appropriately to the right xs * BR value. Extract these from the .json file

Reply:
done